go-controller/pkg/node/node.go

```diff
@@ -652,33 +656,35 @@ func (n *OvnNode) Start(ctx context.Context) error {
 		}
 	}
 
-	// Start the health checking server used by egressip, if EgressIPNodeHealthCheckPort is specified
-	if err := n.startEgressIPHealthCheckingServer(mgmtPortConfig); err != nil {
-		return err
-	}
-
 	klog.Infof("OVN Kube Node initialized and ready.")
 	return nil
 }
 
-func (n *OvnNode) startEgressIPHealthCheckingServer(mgmtPortConfig *managementPortConfig) error {
+func (n *OvnNode) startEgressIPHealthCheckingServer(mgmtPortEntry managementPortEntry) error {
 	healthCheckPort := config.OVNKubernetesFeature.EgressIPNodeHealthCheckPort
 	if healthCheckPort == 0 {
 		klog.Infof("Egress IP health check server skipped: no port specified")
 		return nil
 	}
 
 	var nodeMgmtIP net.IP
-	if mgmtPortConfig.ipv4 != nil {
-		nodeMgmtIP = mgmtPortConfig.ipv4.ifAddr.IP
-	} else if mgmtPortConfig.ipv6 != nil {
-		nodeMgmtIP = mgmtPortConfig.ipv6.ifAddr.IP
-		// Wait for IPv6 address to become usable.
-		if err := ip.SettleAddresses(mgmtPortConfig.ifName, 10); err != nil {
-			return fmt.Errorf("failed start health checking server due to unsettled IPv6: %w", err)
+	var mgmtPortConfig *managementPortConfig = mgmtPortEntry.config
+	// Not all management port interfaces can have IP addresses assignable to them.
+	if mgmtPortEntry.port.HasIpAddr() {
+		if mgmtPortConfig.ipv4 != nil {
+			nodeMgmtIP = mgmtPortConfig.ipv4.ifAddr.IP
+		} else if mgmtPortConfig.ipv6 != nil {
+			nodeMgmtIP = mgmtPortConfig.ipv6.ifAddr.IP
+			// Wait for IPv6 address to become usable.
+			if err := ip.SettleAddresses(mgmtPortConfig.ifName, 10); err != nil {
+				return fmt.Errorf("failed to start Egress IP health checking server due to unsettled IPv6: %w on interface %s", err, mgmtPortConfig.ifName)
+			}
+		} else {
+			return fmt.Errorf("unable to start Egress IP health checking server on interface %s: no mgmt ip", mgmtPortConfig.ifName)
 		}
 	} else {
-		return fmt.Errorf("unable to start health checking server: no mgmt ip")
+		klog.Infof("Skipping interface %s as it should not have an IP address", mgmtPortConfig.ifName)
```

Comment:
NIT: @wizhaoredhat 
s/as it should not have/as it does not have/

Reply:
Thanks, its a bit clearer now!